tests/unit/amo/components/TestUserProfile.js

```diff
@@ -0,0 +1,321 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+
+import UserProfile, { UserProfileBase } from 'amo/components/UserProfile';
+import NotFound from 'amo/components/ErrorPage/NotFound';
+import ReportUserAbuse from 'amo/components/ReportUserAbuse';
+import { fetchUserAccount, getCurrentUser } from 'amo/reducers/users';
+import { createApiError } from 'core/api';
+import { ErrorHandler } from 'core/errorHandler';
+import ErrorList from 'ui/components/ErrorList';
+import LoadingText from 'ui/components/LoadingText';
+import Rating from 'ui/components/Rating';
+import UserAvatar from 'ui/components/UserAvatar';
+import { dispatchSignInActions } from 'tests/unit/amo/helpers';
+import { fakeI18n, shallowUntilTarget } from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  function renderUserProfile({
+    i18n = fakeI18n(),
+    params = { username: 'tofumatt' },
+    store = dispatchSignInActions({
+      userProps: {
+        display_name: 'Matt MacTofu',
+        userId: 500,
+        username: 'tofumatt',
+      },
+    }).store,
+    ...props
+  // eslint-disable-next-line padded-blocks
+  } = {}) {
+
+    return shallowUntilTarget(
+      <UserProfile i18n={i18n} params={params} store={store} {...props} />,
+      UserProfileBase,
+    );
+  }
+
+  function getHeaderPropComponent(root) {
+    const headerContent = root.find('.UserProfile-user-info').prop('header');
+    return shallow(<div>{headerContent}</div>);
+  }
+
+  it('renders user profile page', () => {
+    const root = renderUserProfile();
+
+    expect(root.find('.UserProfile')).toHaveLength(1);
+  });
+
+  it('dispatches fetchUserAccount action if username is not found', () => {
+    const { store } = dispatchSignInActions();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const username = 'i-am-not-in-the-store';
+
+    const root = renderUserProfile({ params: { username }, store });
+
+    sinon.assert.calledWith(dispatchSpy, fetchUserAccount({
+      errorHandlerId: root.instance().props.errorHandler.id,
+      username,
+    }));
+  });
+
+  it('dispatches fetchUserAccount action if username param changes', () => {
+    const { store } = dispatchSignInActions({ username: 'black-panther' });
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const root = renderUserProfile({
+      params: { username: 'black-panther' },
+      store,
+    });
+
+    dispatchSpy.reset();
+
+    sinon.assert.notCalled(dispatchSpy);
```

Comment:
This is not necessary after calling `.reset()`

Reply:
Indeed, I don't need to test sinon in our tests 😅